src/utils/reeds_data_parsing.jl

```diff
@@ -363,6 +387,219 @@ function process_vg(
+        energy_cap = zeros(timesteps_year)
+        for (idx_mon, mon_row) in enumerate(DataFrames.eachrow(monthhours))
+            #@debug reg_plant_subset[findall(x->startswith(mon_row.season,x),reg_plant_subset.szn),:value]
+            try
+                reqd_slice = (mon_row.cumhrs - mon_row.numhrs + 1):(mon_row.cumhrs)
+                monthly_energy[first(reqd_slice)] = (mon_row.numhrs * filter(
+                    x -> (x.month == mon_row.month),
+                    reg_plant_subset,
+                )[
+                    :,
+                    :value,
+                ] * row.MW_sum)[1]
+
+                energy_cap[reqd_slice] .= (mon_row.numhrs * filter(
+                    x -> (x.month == mon_row.month),
+                    reg_plant_subset,
+                )[
+                    :,
+                    :value,
+                ] * row.MW_sum)[1]
+
+                capacity_adjust = hydcapadj[
+                    (hydcapadj.i .== row.i) .&& (hydcapadj.r .== row.r),
+                    [:month, :value],
+                ]
+                dispatch_limit[reqd_slice] .=
+                    (filter(x -> (x.month == mon_row.month), capacity_adjust)[
+                        :,
+                        :value,
+                    ] * row.MW_sum)[1]
+            catch e
+                @error "$(row.r),$(row.i),$(e)"
+            end
+        end
+
+        category = string(row.i)
+        name = "$(category)_$(string(row.r))"
+        region = string(row.r)
+
+        if category in keys(FOR_dict)
+            gen_for = FOR_dict[category]
+        else
+            gen_for = 0.00 # make this 0 for vg if no match
+        end
+
+        # - Charging to genstore is limited by charge_capacity whether from grid or from 
+        # inflows. So, that charge_capacity should be equal to the genflow timeseries.
+        # - Powerflow into grid is limited by grid_injection, which can come from discharge
+        # and/or exogenous inflow. So discharge capacity can be the dispatch limit or 
+        # arbitrarily high, while the grid_injection cap has to the dispatch limit.
+        # - Energy capacity can be arbitrarily high in the absense of reservoir limit to 
+        # ensure month to month energy energy carryover.
+        push!(
+            genstor_array,
+            Gen_Storage(
+                name = name,
+                timesteps = timesteps,
+                region_name = region,
+                charge_cap = repeat(monthly_energy, num_years),
+                discharge_cap = repeat(dispatch_limit, num_years),
+                energy_cap = repeat(energy_cap, num_years),
+                inflow = repeat(monthly_energy, num_years),
+                grid_withdrawl_cap = zeros(timesteps),
+                grid_inj_cap = repeat(dispatch_limit, num_years),
+                type = category,
+                legacy = "New",
+                FOR = 0.0, #TODO: update FORs. For now assuming no outages.
```

Comment:
Looks like this still needs to be addressed. Hydro outage rates are copied from ReEDS-2.0/inputs/plant_characteristics/outage_forced.csv to {ReEDScase}/inputs_case/outage_forced.csv

Reply:
As discussed, I am punting this for later as we do not disaggregate to individual hydro units here and applying FORs might not be appropriate at the regional level.

Reply:
Do we uniformly derate the hydro capacity by the FOR when using the new method? (Did we do that in the old method? Sorry, I'm not sure.) The FOR depends on the hydro type but linearly derating seems most consistent with the treatment of other non-disaggregated techs.
https://github.nrel.gov/ReEDS/ReEDS-2.0/blob/06e282a3011341914f89176e37a09d19d09b4e00/inputs/plant_characteristics/outage_forced.csv#L40-L50

Reply:
No, we do not apply any derating at all